driver-core/src/main/java/com/datastax/driver/core/Connection.java

```diff
@@ -196,6 +196,9 @@ public void operationComplete(ChannelFuture future) throws Exception {
         Futures.addCallback(initFuture, new FutureCallback<Void>() {
             @Override
             public void onSuccess(Void result) {
+                Host host = factory.manager.metadata.getHost(address);
+                if(host != null)
```

Comment:
This happens when the connection succeeds after the node was concurrently removed. I think we need to understand the scenario better to find out what fix is most appropriate.